web-registry/src/lib/ecocredit/api.ts

```diff
@@ -87,10 +88,13 @@ export const getEcocreditsForAccount = async (
 ): Promise<BatchInfoWithBalance[]> => {
   try {
     const batches = await queryEcoBatches();
+    const balancesResponse = await queryEcoBalances(account);
```

Comment:
Single call to get all user credits balance.

Reply:
woohoo!